kotlinsample/src/main/java/com/airbnb/epoxy/kotlinsample/helpers/ViewBindingKotlinModel.kt

```diff
@@ -0,0 +1,57 @@
+package com.airbnb.epoxy.kotlinsample.helpers
+
+import android.view.View
+import androidx.annotation.IdRes
+import androidx.annotation.LayoutRes
+import androidx.viewbinding.ViewBinding
+import com.airbnb.epoxy.EpoxyModel
+import com.airbnb.epoxy.kotlinsample.R
+import java.lang.reflect.Method
+import java.lang.reflect.ParameterizedType
+import java.util.concurrent.ConcurrentHashMap
+import kotlin.properties.ReadOnlyProperty
+import kotlin.reflect.KProperty
+
+/**
+ * A pattern for using epoxy models with Kotlin with no annotations or code generation.
+ *
+ * See [com.airbnb.epoxy.kotlinsample.models.ItemDataClass] for a usage example.
+ */
+abstract class ViewBindingKotlinModel<T : ViewBinding>(
+    @LayoutRes private val layoutRes: Int
+) : EpoxyModel<View>() {
+    // Using reflection to get the static binding method.
+    // Lazy so it's computed only once by instance, when the 1st ViewHolder is actually created.
+    private val actualTypeOfThis by lazy { this::class.java.genericSuperclass as ParameterizedType }
+    private val kClass by lazy { (actualTypeOfThis.actualTypeArguments[0] as Class<ViewBinding>) }
+    private val bindingMethod by lazy { getBindMethod(kClass)!! }
```

Comment:
as an optimization I would:
1. make `kClass` accessed as a function, since the two `lazy` calls are slightly expensive and we only need to access it once
2. have the map key be `this::class`, falling back to calculate `kclass` on a cache miss to calculate the binding method. this prevents all the reflection in most cases

Reply:
Since this::class is the naw key, I moved the reflection to the getOrPut. Is it ok?